lib/collection/src/config.rs

```diff
@@ -179,15 +178,14 @@ impl CollectionParams {
                     VectorDataConfig {
                         size: params.size.get() as usize,
                         distance: params.distance,
-                        hnsw_config: params
-                            .hnsw_config
-                            .and_then(|c| c.update(collection_hnsw).ok()),
-                        quantization_config: params
-                            .quantization_config
-                            .as_ref()
+                        // Plain (disabled) index
+                        index: Indexes::Plain {},
+                        // Only set if enabled on segment level as well, prefer vector level config
+                        quantization_config: collection_quantization
+                            .and(params.quantization_config.as_ref())
                             .or(collection_quantization)
                             .cloned(),
```

Comment:
Should we pick enable and pick quantization settings at all times? Or should we only enable quantization through a segment optimizer? This currently assumes the first.

This assumes quantization should **only** be used if it is enabled on the segment level as well.

Reply:
In a call we discussed it should be the second option. Quantization should still be used for a vector if it is not configured on segment level.

Changes have been made accordingly.